EventListener/ReportSubscriber.php

```diff
@@ -0,0 +1,283 @@
+use Mautic\ReportBundle\Event\ReportBuilderEvent;
+use Mautic\ReportBundle\Event\ReportGeneratorEvent;
+use Mautic\ReportBundle\ReportEvents;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompaniesSegments;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompanySegment;
+use MauticPlugin\LeuchtfeuerCompanySegmentsBundle\Entity\CompanySegmentRepository;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ReportSubscriber implements EventSubscriberInterface
+{
+    public const CONTEXT_COMPANY_SEGMENTS         = CompanySegment::TABLE_NAME;
+    public const COMPANY_TABLE                    = 'companies';
+    public const COMPANIES_PREFIX                 = 'comp';
+    public const COMPANY_SEGMENTS_XREF_PREFIX     = 'csx';
+    public const COMPANY_SEGMENTS_XREF_TABLE      = CompaniesSegments::TABLE_NAME;
+
+    public function __construct(
+        private CompanyReportData $companyReportData,
+        private CompanySegmentRepository $companySegmentRepository,
+        private Connection $db,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ReportEvents::REPORT_ON_BUILD    => ['onReportBuilder', 0],
+            ReportEvents::REPORT_ON_GENERATE => ['onReportGenerate', 0],
+        ];
+    }
+
+    public function onReportBuilder(ReportBuilderEvent $event): void
+    {
+        if (!$event->checkContext([self::CONTEXT_COMPANY_SEGMENTS])) {
+            return;
+        }
+
+        $keys = [
+            'comp.id',
+            'comp.companyaddress1',
+            'comp.companyaddress2',
+            'comp.companyemail',
+            'comp.companyphone',
+            'comp.companycity',
+            'comp.companystate',
+            'comp.companyzipcode',
+            'comp.companycountry',
+            'comp.companyname',
+            'comp.companywebsite',
+            'comp.companynumber_of_employees',
+            'comp.companyfax',
+            'comp.companyannual_revenue',
+            'comp.companyindustry',
+            'comp.companydescription',
+        ];
+        $columns         = $this->companyReportData->getCompanyData();
+        $filteredColumns = array_intersect_key($columns, array_flip($keys));
+
+        $segmentList = $this->getFilterSegments();
+
+        $segmentFilter = [self::COMPANY_SEGMENTS_XREF_PREFIX.'.segment_id' => [
+            'alias'     => 'companysegments',
+            'label'     => 'mautic.company_segments.report.company_segments',
+            'type'      => 'select',
+            'list'      => $segmentList,
+            'operators' => [
+                'in'       => 'mautic.core.operator.in',
+                'notIn'    => 'mautic.core.operator.notin',
+                'empty'    => 'mautic.core.operator.isempty',
+                'notEmpty' => 'mautic.core.operator.isnotempty',
+            ],
+        ],
+        ];
+        $filters = array_merge($filteredColumns, $segmentFilter);
```

Comment:
It's much more efficient to use a regular `$filteredColumns[self::COMPANY_SEGMENTS_XREF_PREFIX.'.segment_id'] = [...`